ci-operator/step-registry/openshift/e2e/gcp/libvirt/cert-rotation/conf/openshift-e2e-gcp-libvirt-cert-rotation-conf-commands.sh

```diff
@@ -0,0 +1,440 @@
+  # happen if time was progressing naturally. Kill image-prune jobs here.
+  oc delete jobs --all -n openshift-image-registry
+  sleep 10
+  while ! oc wait co --all --for='condition=Degraded=False' --timeout=10s && [ $retries -lt 50 ]; do
+    (( retries++ ))
+  done
+}
+
+checkProgressingCOs () {
+  retries=0
+  # image-pruner job in openshift-image-registry namespace may be stuck due to time skew. This would not
+  # happen if time was progressing naturally. Kill image-prune jobs here.
+  oc delete jobs --all -n openshift-image-registry
+  sleep 10
+  while ! oc wait co --all --for='condition=Progressing=False' --timeout=10s && [ $retries -lt 50 ]; do
+    jumpstartNodes
+    (( retries++ ))
+  done
+}
+
+checkAvailableCOs () {
+  retries=0
+  while ! oc wait co --all --for='condition=Available=True' --timeout=20s && [ $retries -lt 100 ]; do
+    jumpstartNodes
+    (( retries++ ))
+  done
+}
+
+sudo systemctl stop chronyd
+
+SKEW=${1:-+1y}
+
+OC=${OC:-oc}
+SSH=${SSH:-ssh}
+
+masters=$( ${OC} get nodes --selector='node-role.kubernetes.io/master' --template='{{ range $index, $_ := .items }}{{ range .status.addresses }}{{ if (eq .type "InternalIP") }}{{ if $index }} {{end }}{{ .address }}{{ end }}{{ end }}{{ end }}' )
+workers=$( ${OC} get nodes --selector='!node-role.kubernetes.io/master' --template='{{ range $index, $_ := .items }}{{ range .status.addresses }}{{ if (eq .type "InternalIP") }}{{ if $index }} {{end }}{{ .address }}{{ end }}{{ end }}{{ end }}' )
+
+function run-on {
+        for n in ${1}; do ${SSH} core@"${n}" sudo 'bash -eEuxo pipefail' <<< ${2}; done
+}
+
+ssh-keyscan -H ${masters} ${workers} >> ~/.ssh/known_hosts
+
+run-on "${masters} ${workers}" "systemctl stop kubelet"
+
+# Destroy all containers on workers.
+run-on "${workers}" "crictl rm --all -f"
+# Destroy all containers on masters except KAS and etcd.
+run-on "${masters}" '
+kas_id=$( crictl ps --name="^kube-apiserver$" -q )
+[[ -n "${kas_id}" ]]
+etcd_id=$( crictl ps --name="^etcd$" -q )
+[[ -n "${etcd_id}" ]]
+other_ids=$( crictl ps --all -q | ( grep -v -e "${kas_id}" -e "${etcd_id}" || true ) )
+if [ -n "${other_ids}" ]; then
+        crictl rm -f ${other_ids}
+fi;
+'
+
+# Delete all pods, especialy the operators. Makes sure it needs KCM and KS working when starting again.
+${OC} delete pods -A --all --force --grace-period=0 --timeout=0
+
+# Destroy the remaining containers on masters
+run-on "${masters}" "crictl rm --all -f"
+
+run-on "${masters} ${workers}" "systemctl disable chronyd --now"
+
+# Set time only as a difference to the synced time so we don't introduce a skew between the machines which would break etcd, leader election and others.
+run-on "${masters} ${workers}" "
+timedatectl status
+timedatectl set-ntp false
+timedatectl set-time '${SKEW}'
+timedatectl status
+"
+
+# now set date for host
+timedatectl set-time ${SKEW} 
+
+run-on "${masters} ${workers}" "systemctl start kubelet"
```

Comment:
nit: move it above https://github.com/openshift/release/pull/12878/files#diff-d2dafc1415630d370225f3b8a10394b14e8d13d0f5d9f7056413251863d73eeeR403 so it is logically grouped with modifying the nodes

Reply:
If i recall, I had to start kubelet _after_ setting time on the host? I might be wrong, I'll give it a try

Reply:
you recall correctly, the line above is not doing that though. it sets the time on the bastion or wherever your run the script from, not on the master with the kubelet.